### PR TITLE
Puppy Camera for 2.1

### PR DESCRIPTION
**Pull request by FazanaJ**

### CODE REVIEW

File: include/text_strings.h.in
Comment (thecozies) [inline include/text_strings.h.in:29]: can you remove the unused strings?

File: src/game/puppycam2.c
Comment (thecozies) [inline src/game/puppycam2.c:547]: this shouldn't pollute the logs, if it's a convenient log to have in here i would comment it out or have a define or macro at the top of this file for cam debug

Comment (FazanaJ) [reply src/game/puppycam2.c:547]: oops dev debugging leftover

File: src/game/puppycam2.c
Comment (thecozies) [inline src/game/puppycam2.c:615]: input type defines

File: src/game/puppycam2.c
Comment (thecozies) [inline src/game/puppycam2.c:1204]: there are quite a lot of these angle approach function, is there an existing function you could use or create a new one?

**Review by thecozies — COMMENTED** [review]
few small things for now, gonna need to test it out and see

File: include/text_strings.h.in
Comment (thecozies) [inline include/text_strings.h.in:28]: allons-y

File: include/text_strings.h.in
Comment (thecozies) [inline include/text_strings.h.in:44]: delet

**Comment by thecozies** [comment]
classic mode:

"C Up" view is hard to use, super sensitive Y and looking down shouldnt zoom out imo

https://user-images.githubusercontent.com/79979276/207616056-917ca429-4823-4797-b888-5c825e03d9af.mp4

**Comment by thecozies** [comment]
mode 2, C up feels weird, should be closer to marios head instead of dragging against the ground, you can see little jumps in movement


https://user-images.githubusercontent.com/79979276/207617073-6dff7e0d-185d-4a7d-bf79-f6f905e396e3.mp4

**Comment by thecozies** [comment]
While moving, holding C Up makes mario disappear


https://user-images.githubusercontent.com/79979276/207617143-1f1634a9-f729-448a-9419-1e6a1c30d750.mp4


I would try to make this point the focus, pull back and keep mario in view
image

**Comment by thecozies** [comment]
didnt puppycam previously pan in the direction you're moving or facing? it does in lucys but it seems non existent now

https://user-images.githubusercontent.com/79979276/207620907-2d2f2a52-2281-49c9-82be-b57589a8fcbc.mp4

File: include/text_strings.h.in
Comment (gheskett) [inline include/text_strings.h.in:8]: I'd maybe replace all tabs with spaces in this file for consistency.

File: src/game/puppycam2.c
Comment (gheskett) [inline src/game/puppycam2.c:88]: What is "Option Name" supposed to be? I would either sort these to make sense or replace them with enums; this struct is a bit confusingly laid out in general.

File: src/game/puppycam2.c
Comment (gheskett) [inline src/game/puppycam2.c:134]: spacing

File: src/game/puppycam2.c
Comment (gheskett) [inline src/game/puppycam2.c:488]: Maybe this has been addressed in Puppyprint already, but `append_puppyprint_log`'s actual definition should be ifdef'd, not the calls to it.

File: src/game/puppycam2.c
Comment (gheskett) [inline src/game/puppycam2.c:504]: Wat

**Review by gheskett — CHANGES_REQUESTED** [review]

File: enhancements/puppycam_angles.inc.c
Comment (arthurtilly) [reply enhancements/puppycam_angles.inc.c:1]: reonu told me i was allowed to delete the folder after reonucam patch got yeeted :pensive:

File: src/game/puppycam2.c
Comment (FazanaJ) [reply src/game/puppycam2.c:513]: backwards compatability stuff, completely unrelated to current nodes

File: enhancements/puppycam_angles.inc.c
Comment (FazanaJ) [reply enhancements/puppycam_angles.inc.c:1]: fast64 relies on this folder existing when exporting PC1 nodes, so it must remain for a while longer.

File: src/game/puppycam2.c
Comment (FazanaJ) [reply src/game/puppycam2.c:497]: I'll leave this as is for now, and make this change in Puppyprint 2.1.

File: enhancements/puppycam_angles.inc.c
Comment (gheskett) [reply enhancements/puppycam_angles.inc.c:1]: @thecozies your input is desirable here

Comment (thecozies) [reply enhancements/puppycam_angles.inc.c:1]: I think that this is a file, and it is inside of the enhancements folder

Comment (thecozies) [reply enhancements/puppycam_angles.inc.c:1]: https://github.com/Fast-64/fast64/blob/6dececc9f42bd4b7632c84ac4b1b34fbc22bcf79/fast64_internal/sm64/sm64_level_writer.py#L989

Comment (gheskett) [reply enhancements/puppycam_angles.inc.c:1]: Uhh, right. What do we need to do to destroy this folder lol

Comment (thecozies) [reply enhancements/puppycam_angles.inc.c:1]: it does require fast64 change but it'd be a one liner

Comment (FazanaJ) [reply enhancements/puppycam_angles.inc.c:1]: If fast64 is updated to use a different directory then I can easily move this file to a more sensible location

Comment (thecozies) [reply enhancements/puppycam_angles.inc.c:1]: let's go ahead and move this somewhere better and I can fix that soon

Comment (FazanaJ) [reply enhancements/puppycam_angles.inc.c:1]: ok my first idea is /include, otherwise we just shove it in /src/game

Comment (gheskett) [reply enhancements/puppycam_angles.inc.c:1]: Tbh I think this and all other puppy stuffs should be moved to their own subfolder. `src/game` is getting pretty chonky.

Just in general, we need more subfolders tbh.

Comment (arthurtilly) [reply enhancements/puppycam_angles.inc.c:1]: personally i think we have too many subfolders already

Comment (gheskett) [reply enhancements/puppycam_angles.inc.c:1]: We have to strike a balance between files, directories, and file sizes in general. I think we have too many standalone files in `src/game`, though I can see it being unreasonable to actually categorize everything properly without hella merge conflicts.

Comment (arthurtilly) [reply enhancements/puppycam_angles.inc.c:1]: my vote is to try to balance them out, maybe combine s2d/usb/hvqm/etc and the puppy stuff into one subfolder for custom libraries. that way we reduce src/ subfolders and also reduce src/game files

**Review by thecozies — COMMENTED** [review]
collision is a little weird here (also noticed a debug print on the bottom left)

https://user-images.githubusercontent.com/79979276/215809765-122bd73e-59e6-4176-8121-008886e7620c.mp4

___

i also noticed that panning is baaaarely there, like in my obs screen cap you can see where the center of the screen is and mario is like "one mario" to the right of the center

image

**Comment by thecozies** [comment]
first person might not be precise enough, you can see how jittery it is when looking around

https://user-images.githubusercontent.com/79979276/215812017-87cfbcc7-e62a-4be9-909e-247db3b2cec5.mp4

**Comment by FazanaJ** [comment]
> first person might not be precise enough, you can see how jittery it is when looking around

Fixed.

Will look into panning at a future time

File: src/game/puppycam2.c
Comment (gheskett) [reply src/game/puppycam2.c:497]: This was not addressed in Puppyprint 2.1; please fix.

Comment (gheskett) [reply src/game/puppycam2.c:497]: This conditional check should be removed now.

Also I just noticed `#if`. Double check and make sure those don't exist anywhere else in here.

File: include/text_strings.h.in
Comment (gheskett) [reply include/text_strings.h.in:8]: Can you fix this one too?

**Review by gheskett — APPROVED** [review]
Approved, still want the approval of cozies before this gets merged.

**Review by thecozies — APPROVED** [review]
approval is
# now

**Comment by CrashOveride95** [comment]
N

**Pull request merged**
